suro-core/src/main/java/com/netflix/suro/sink/ThreadPoolQueuedSink.java

```diff
@@ -40,7 +40,7 @@ public boolean offer(Runnable runnable) {
                 maxPoolSize == 0 ? 10 : maxPoolSize,
                 10, TimeUnit.SECONDS,
                 jobQueue,
-                new ThreadFactoryBuilder().setNameFormat(threadFactoryName + "-Sender-%d").build());
+                new ThreadFactoryBuilder().setNameFormat(threadFactoryName + "-Sender-%d").setDaemon(true).build());
```

Comment:
Set threads as daemon is generally not recommended for I/O: http://stackoverflow.com/questions/2213340/what-is-daemon-thread-in-java

Calling ```innerClose()``` will at least wait for some time for the sink to get drained. So I think invoking ```close()``` method at shutdown can reduce the data loss. We can either use shutdown hook to call ```close()``` or annotate ```close()``` with ```@PreDestroy``` in DI mode.